### PR TITLE
Feature/unr 1480 batch spatial position updates

### PR DESCRIPTION
**Pull request by m-samiec**
#### Description
Add option to batch spatial position updates to runtime. Default on.

#### Release note
Feature: Batch spatial position updates to runtime

#### Tests
Ran in AI Gym.

#### Primary reviewers
@samiwh @mattyoung-improbable

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp [hunk 1]
```diff
@@ -627,6 +627,24 @@ void USpatialSender::RetryReliableRPC(TSharedRef<FReliableRPCForRetry> RetryRPC)
...
+	ChannelsToUpdatePosition.Add(Channel);
+}
+
+void USpatialSender::ProcessPositionUpdates()
+{
+	for (auto& Channel : ChannelsToUpdatePosition)
+	{
+		if (Channel.IsValid())
+		{
+			Channel->UpdateSpatialPosition( /*bCheckLastUpdateTime*/ false );
```
Comment (mattyoung-improbable) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp]: Why don't we want the threshold checks to happen if batching?

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 2]
```diff
@@ -1201,6 +1212,16 @@ void USpatialNetDriver::TickFlush(float DeltaTime)
 		TimerManager.Tick(DeltaTime);
 	}
 
+	if (GetDefault<USpatialGDKSettings>()->bBatchSpatialPositionUpdates)
+	{
+		if ((Time - TimeWhenPositionLastUpdated) >= (1.0f / GetDefault<USpatialGDKSettings>()->PositionUpdateFrequency))
+		{
+			TimeWhenPositionLastUpdated = Time;
+
+			Sender->ProcessPositionUpdates();
```
Comment (samiwh) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Do we want to batch all position updates together? Spiky traffic tends to have it's own detrimental affect (although I don't have a good intuition for what magnitude).

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp [hunk 3]
```diff
@@ -748,10 +755,12 @@ void USpatialActorChannel::OnCreateEntityResponse(const Worker_CreateEntityRespo
 	UE_LOG(LogSpatialActorChannel, Verbose, TEXT("Created entity (%lld) for: %s."), Op.entity_id, *Actor->GetName());
 }
 
-void USpatialActorChannel::UpdateSpatialPosition()
+void USpatialActorChannel::UpdateSpatialPosition(bool bCheckLastUpdateTime)
```
Comment (samiwh) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp]: This is only ever called with a literal argument, and it the affect is limited to a check at the beginning of the function. Would it be better to split this into two methods, one of which calls the other if appropriate.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 2 above)
Comment (samiwh) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: That said: if we've measured that this is an improvement for the entity numbers we care about then great and we can make this more complicated later if those entity numbers increase.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp (hunk 3 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp]: Yep - good point - I'll look into it.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 2 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Yes, I think we deal with that potential optimisation separately. You can turn off the batching if it's problematic.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 4]
```diff
@@ -39,6 +39,17 @@ DEFINE_LOG_CATEGORY(LogSpatialOSNetDriver);
 DECLARE_CYCLE_STAT(TEXT("ServerReplicateActors"), STAT_SpatialServerReplicateActors, STATGROUP_SpatialNet);
 DEFINE_STAT(STAT_SpatialConsiderList);
 
+USpatialNetDriver::USpatialNetDriver(const FObjectInitializer& ObjectInitializer)
+	: Super(ObjectInitializer)
+	, bAuthoritativeDestruction(false)
```
Comment (samiwh) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Possible I've misunderstood, but this is always set to true in the init function. Is that function always called?

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: It is always called, I just added a initializer list to default to sane values. I'll switch to true.

**Pull request merged**